app/views/auth_source_ldaps/_form.html.erb

```diff
@@ -40,4 +40,13 @@
   </div>
 
   <%= submit_or_cancel f %>
+  <button id="aid_test_connection_button_auth_source_ldap" class="btn btn-info" data-id="aid_test_connection_auth_source_ldap" type="button" style="margin-left:3px;" onclick="test_connection('<%= url_for(:controller => 'auth_source_ldaps', :action=>'test_connection') %>');">
+    <span class="glyphicon glyphicon-random"></span>
+    <span> <%= _('Test Connection') %> </span>
+  </button>
+  <script type="text/javascript">
+    // A more cleaner implementation will definitely listen to an event emitted when a two-pane page is done loading
```

Comment:
Is this just to put the button next to Submit?  I wonder if that's the best place for it, e.g. the compute resource edit page puts it inside the form.

I think you can put this in the ContentLoad event otherwise...